json.md

```diff
@@ -0,0 +1,415 @@
+* **inputs**, which are files the build action is performed on.
+
+We'll create our build action file in `<project_root>/tool/build_actions.dart`.
+
+**tool/build_actions.dart**
+
+<!-- skip -->
+{% prettify dart %} 
+import 'package:build_runner/build_runner.dart';
+import 'package:json_serializable/json_serializable.dart';
+
+List<BuildAction> get buildActions => [
+      new BuildAction(
+        // The builder to run on our source files.
+        jsonPartBuilder(),
+
+        // The name of the current package, usually the same as the name 
+        // for your Flutter project. For example, if you created a new
+        // project with "flutter run hello", your package name here would
+        // be 'hello'.
+        '[[highlight]]<your_package_name_here>[[/highlight]]',
+
+        // All of the files this `BuildAction` should target when run.
+        inputs: const [
+          'lib/*.dart',
+        ],
+      ),
+    ];
+{% endprettify %}
+
+We are declaring a top-level variable called `buildActions`, which returns a
+`List` containing all build actions in our project. In this case, we only have
+one build action, which generates JSON serialization logic for files in the
+_lib_ folder.
+
+It is good practice to define build actions in the same file, as they are used
+by the _build_ and the _watcher_ files.
+
+Which brings us to the next topic, the build file.
+
+#### Creating the build file
+
+Our build file runs the actions responsible for generating JSON serialization
+logic for our classes. When run, it triggers the build actions, searches for all
+the relevant files, generates the necessary code and then complete.
+
+We'll create the build file in `<project_root>/tool/build.dart`.
+
+**tool/build.dart**
+
+<!-- skip -->
+```dart
+import 'dart:io';
+import 'package:build_runner/build_runner.dart';
+import 'build_actions.dart';
+
+main() async {
+  var result = await build(buildActions, deleteFilesByDefault: true);
+  if (result.status == BuildStatus.failure) {
+    exitCode = 1;
+  }
+}
+```
+
+Now we can generate all the necessary JSON serialization source files by running
+`dart tool/build.dart` in our project's root folder. While the build file is all
+we need for one-time source code generation, it would be nice if we did not have
+to run it manually every time we make changes.
+
+#### Creating the watcher file
+
+A _watcher_ makes our source code generation progress more convenient. It
+watches changes in our project and automatically builds the necessary files when
+needed.
+
+The watcher file also lives in the tool folder, so we create it in
+`<project_root>/tool/watch.dart`.
+
+**tool/watch.dart**
+
+<!-- skip -->
+```dart
+import 'package:build_runner/build_runner.dart';
+import 'build_actions.dart';
+
+main() {
+  watch(buildActions, deleteFilesByDefault: true);
+}
+```
+
+We can run the watcher by doing `dart tool/watch.dart` in our project's root
+folder. It is safe to start the watcher once and leave it running in the
```

Comment:
please provide an example of how to run this. (it's not obvious how to use the command-line Dart VM that is bundled with Flutter.)